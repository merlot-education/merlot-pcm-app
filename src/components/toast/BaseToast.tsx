import React from 'react'
import { View, Text, useWindowDimensions, StyleSheet } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

import {
  TextTheme,
  borderRadius,
  borderWidth,
  ColorPallet,
} from '../../theme/theme'

interface BaseToastProps {
  title: string
  body: string
  toastType: string
}

export enum ToastType {
  Success = 'success',
  Info = 'info',
  Warn = 'warn',
  Error = 'error',
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'flex-start',
    flexDirection: 'row',
    borderWidth,
    borderRadius,
  },
  textContainer: {
    flexShrink: 1,
    marginVertical: 10,
    marginRight: 10,
  },
  icon: {
    marginTop: 15,
    marginHorizontal: 15,
  },
  title: {
    fontWeight: 'bold',
  },
  body: {
    marginTop: 10,
  },
})

const BaseToast: React.FC<BaseToastProps> = ({ title, body, toastType }) => {
  const { width } = useWindowDimensions()
  const iconSize = 24
  let iconName = ''
  let backgroundColor = ''
  let borderColor = ''
  let iconColor = ''
  let textColor = ''

  switch (toastType) {
    case ToastType.Success:
      iconName = 'check-circle'
      backgroundColor = ColorPallet.notification.success
      borderColor = ColorPallet.notification.successBorder
      iconColor = ColorPallet.notification.successIcon
      textColor = ColorPallet.notification.successText
      break

    case ToastType.Info:
      iconName = 'info'
      backgroundColor = ColorPallet.notification.info
      borderColor = ColorPallet.notification.infoBorder
      iconColor = ColorPallet.notification.infoIcon
      textColor = ColorPallet.notification.infoText
      break

    case ToastType.Warn:
      iconName = 'report-problem'
      backgroundColor = ColorPallet.notification.warn
      borderColor = ColorPallet.notification.warnBorder
      iconColor = ColorPallet.notification.warnIcon
      textColor = ColorPallet.notification.warnText
      break

    case ToastType.Error:
      iconName = 'error'
      backgroundColor = ColorPallet.notification.error
      borderColor = ColorPallet.notification.errorBorder
      iconColor = ColorPallet.notification.errorIcon
      textColor = ColorPallet.notification.errorText
      break

    default:
      throw new Error('ToastType was not set correctly.')
  }

  return (
    <View
      style={[
        styles.container,
        { backgroundColor, borderColor, width: width - width * 0.1 },
      ]}
    >
      <Icon
        style={styles.icon}
        name={iconName}
        color={iconColor}
        size={iconSize}
      />
      <View style={[styles.textContainer]}>
        <Text style={[TextTheme.normal, styles.title, { color: textColor }]}>
          {title}
        </Text>
        <Text style={[TextTheme.label, styles.body, { color: textColor }]}>
          {body}
        </Text>
      </View>
    </View>
  )
}

export default BaseToast
