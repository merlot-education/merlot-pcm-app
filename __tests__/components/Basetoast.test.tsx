import React from 'react'
import { create } from 'react-test-renderer'

import BaseToast, { ToastType } from '../../src/components/toast/BaseToast'

describe('BaseToast Component', () => {
  it('Info renders correctly', () => {
    const tree = create(
      <BaseToast
        title="Info"
        body="This is info message"
        toastType={ToastType.Info}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Success renders correctly', () => {
    const tree = create(
      <BaseToast
        title="Success"
        body="This is Success message"
        toastType={ToastType.Success}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Warn renders correctly', () => {
    const tree = create(
      <BaseToast
        title="Warn"
        body="This is warning message"
        toastType={ToastType.Warn}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Error renders correctly', () => {
    const tree = create(
      <BaseToast
        title="Error"
        body="This is error message"
        toastType={ToastType.Error}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
