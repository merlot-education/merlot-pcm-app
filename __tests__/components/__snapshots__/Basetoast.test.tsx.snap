// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseToast Component Error renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "borderRadius": 4,
        "borderWidth": 2,
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#ECC8C4",
        "borderColor": "#B25B59",
        "width": 675,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#B25B59",
          "fontSize": 24,
        },
        Object {
          "marginHorizontal": 15,
          "marginTop": 15,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flexShrink": 1,
          "marginRight": 10,
          "marginVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 18,
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "color": "#B25B59",
          },
        ]
      }
    >
      Error
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          Object {
            "marginTop": 10,
          },
          Object {
            "color": "#B25B59",
          },
        ]
      }
    >
      This is error message
    </Text>
  </View>
</View>
`;

exports[`BaseToast Component Info renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "borderRadius": 4,
        "borderWidth": 2,
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#CCE8F4",
        "borderColor": "#4F7C9E",
        "width": 675,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#4F7C9E",
          "fontSize": 24,
        },
        Object {
          "marginHorizontal": 15,
          "marginTop": 15,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flexShrink": 1,
          "marginRight": 10,
          "marginVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 18,
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "color": "#4F7C9E",
          },
        ]
      }
    >
      Info
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          Object {
            "marginTop": 10,
          },
          Object {
            "color": "#4F7C9E",
          },
        ]
      }
    >
      This is info message
    </Text>
  </View>
</View>
`;

exports[`BaseToast Component Success renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "borderRadius": 4,
        "borderWidth": 2,
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#D2EBCA",
        "borderColor": "#60775A",
        "width": 675,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#60775A",
          "fontSize": 24,
        },
        Object {
          "marginHorizontal": 15,
          "marginTop": 15,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flexShrink": 1,
          "marginRight": 10,
          "marginVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 18,
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "color": "#60775A",
          },
        ]
      }
    >
      Success
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          Object {
            "marginTop": 10,
          },
          Object {
            "color": "#60775A",
          },
        ]
      }
    >
      This is Success message
    </Text>
  </View>
</View>
`;

exports[`BaseToast Component Warn renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "borderRadius": 4,
        "borderWidth": 2,
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#F7F3D7",
        "borderColor": "#AE8E56",
        "width": 675,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#AE8E56",
          "fontSize": 24,
        },
        Object {
          "marginHorizontal": 15,
          "marginTop": 15,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "flexShrink": 1,
          "marginRight": 10,
          "marginVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 18,
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "color": "#AE8E56",
          },
        ]
      }
    >
      Warn
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#1E9BF2",
            "fontSize": 14,
            "fontWeight": "bold",
          },
          Object {
            "marginTop": 10,
          },
          Object {
            "color": "#AE8E56",
          },
        ]
      }
    >
      This is warning message
    </Text>
  </View>
</View>
`;
